.circleci/config.yml

```diff
@@ -93,7 +93,7 @@ jobs:
         enum: [ development, staging, production ]
     environment:
       SAM_CONFIG_FILE: samconfig.toml.d/ci-<<parameters.dc-environment>>.toml
-      SAM_LAMBDA_CONFIG_ENV: <<parameters.dc-environment>>
+      DC_ENVIRONMENT: <<parameters.dc-environment>>
       SAM_PUBLIC_CONFIG_ENV: <<parameters.dc-environment>>-public-access
```

Comment:
SAM_PUBLIC_CONFIG_ENV isn't found anywhere else in the project, so I don't know what it's for? I was wondering about either renaming it to DC_ENVIRONMENT_PUBLIC for consistency, or simply removing it, but I didn't because I wasn't sure if CircleCI needs it for some reason..

Reply:
As far as I'm aware the variable `SAM_LAMBDA_CONFIG_ENV` is not special. It is just a name we picked and we can change it to `DC_ENVIRONMENT` to align with other projects with no consequences. As long as we update the one place we're referencing it in `base.py`, we'll be fine. Maybe @symroe can double check me on that.

Reply:
This is used [here](https://github.com/DemocracyClub/Website/blob/master/.circleci/config.yml#L113) to pick the config in [this directory](https://github.com/DemocracyClub/Website/tree/master/samconfig.toml.d).

DC environment is an ok proxy for this in the way that we're using it currently, but the initial intend of this was to allow us to have a more diverse array of deployments. 

For example, the person who made this system thought that devs might each have their own deployment environment for longer running feature dev. Or, in fact, we might have an environment per feature, whatever.

This means we might have had `DC_ENVIRONMENT=development && SAM_LAMBDA_CONFIG_ENV=feature-new-design` (where `feature-new-design` was defined in `samconfig.toml.d`).

As I say, we're not using this at the moment, so I think I'm ok with combining the concepts, but there was a reason for this in the past. This is also a good example in prematurely adding features. The contractor who did this had a good reason though: we were kicking the tyres of a new system and wanted to see what it could do.

Reply:
also this PR is already updating

https://github.com/DemocracyClub/Website/blob/179b497982d90f4fe4e574e272184227750d4828/.circleci/config.yml#L113

as well as

https://github.com/DemocracyClub/Website/blob/179b497982d90f4fe4e574e272184227750d4828/democracy_club/settings/base.py#L239

Reply:
Sorry, to clarify I was asking about the variable on line 97: `SAM_PUBLIC_CONFIG_ENV` (PUBLIC rather than LAMBDA) which I haven't changed but was wondering if I should in order to align in with the change from `SAM_LAMBDA_CONFIG_ENV` to `DC_ENVIRONMENT`.

Reply:
Right yes sorry.

I think the `SAM_PUBLIC_CONFIG_ENV` variable is not used anywhere in this project and we can just delete it. I reckon it is just a copy & paste from another project (the devs.DC API) that does use it for something.